calico_rkt/calico_rkt.py

```diff
@@ -1,198 +1,212 @@
     """
     # Find the endpoint ID. We need this to find any ACL rules
     try:
-        endpoint = client.get_endpoint(
-            hostname=hostname,
-            orchestrator_id=orchestrator_id,
-            workload_id=container_id
-        )
+        endpoint = _datastore_client.get_endpoint(hostname=hostname,
+                                                  orchestrator_id=orchestrator_id,
+                                                  workload_id=container_id)
     except KeyError:
-        print_stderr("Container %s doesn't contain any endpoints" % container_id)
+        _log.error(
+            "Container %s doesn't contain any endpoints" % container_id)
         sys.exit(1)
 
     # Remove any IP address assignments that this endpoint has
     for net in endpoint.ipv4_nets | endpoint.ipv6_nets:
         assert(net.size == 1)
-        client.unassign_address(None, net.ip)
+        _datastore_client.unassign_address(None, net.ip)
 
     # Remove the endpoint
     netns.remove_veth(endpoint.name)
 
     # Remove the container from the datastore.
-    client.remove_workload(hostname, orchestrator_id, container_id)
+    _datastore_client.remove_workload(hostname=hostname,
+                                      orchestrator_id=orchestrator_id,
+                                      workload_id=container_id)
 
-    print_stderr("Removed Calico interface from %s" % container_id)
+    _log.info("Removed Calico interface from %s" % container_id)
 
-def _create_profile(endpoint, profile_name, ip, client):
-    """
-    Configure the calico profile for a pod.
 
-    Currently assumes one pod with each name.
+def _set_profile_on_endpoint(endpoint, profile_name, ip):
     """
-    print_stderr('Configuring Pod Profile: %s' % profile_name)
+    Configure the calico profile to the endpoint
+    """
+    _log.info('Configuring Pod Profile: %s' % profile_name)
 
-    if client.profile_exists(profile_name):
-        print_stderr("Profile with name %s already exists, applying to endpoint." % (profile_name))
+    if _datastore_client.profile_exists(profile_name):
+        _log.info(
+            "Profile with name %s already exists, applying to endpoint." % (profile_name))
 
     else:
-        client.create_profile(profile_name)
-        # _apply_rules(profile_name, client)
+        _log.info("Creating profile %s." % (profile_name))
+        _datastore_client.create_profile(profile_name)
+        # _apply_default_rules(profile_name)
 
     # Also set the profile for the workload.
-    print_stderr('Setting profile %s on endpoint %s' %
-                 (profile_name, endpoint.endpoint_id))
-    client.set_profiles_on_endpoint(
-        [profile_name], endpoint_id=endpoint.endpoint_id
-    )
-    print_stderr("Finished configuring profile.")
+    _datastore_client.set_profiles_on_endpoint(profile_names=[profile_name],
+                                               endpoint_id=endpoint.endpoint_id)
+
     dump = json.dumps(
```

Comment:
Passing in the IP just to print it out doesn't seem sensible since set_profile_on_endpoint doesn't use the IP for anything else - try moving this up into the caller (create)